### PR TITLE
Fix drag and drop support on linux (and macOS?) with SDL and .NET core

### PR DESCRIPTION
**Pull request by morguldir**
### Purpose of this PR

Make Drag and drop support work when running on non-unicode platforms with SDL because .NET Core ~was too lazy to do it~ handles text encoding differently.

Resolves https://github.com/ppy/osu/issues/3444

### CODE REVIEW

**Comment by peppy** [comment]
Thanks for submitting this. Any idea how it regressed, since it seems to have been working up to a certain point.

**Comment by morguldir** [comment]
It was working when SDL wasn’t loading and we were falling back to X11, which #27 fixed

File: src/OpenTK/Platform/SDL2/Sdl2NativeWindow.cs [hunk 1]
```diff
@@ -303,9 +303,16 @@ private static void ProcessMouseWheelEvent(Sdl2NativeWindow window, MouseWheelEv
             window.OnMouseWheel(ev.X, ev.Y, false);
         }
 
+        private static readonly int SystemDefaultCharSize = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? 2 : 1;
```
Comment (VPeruS) [inline src/OpenTK/Platform/SDL2/Sdl2NativeWindow.cs]:
```suggestion
        private static readonly bool IsFilenameUniEncoded = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
```

File: src/OpenTK/Platform/SDL2/Sdl2NativeWindow.cs [hunk 2]
```diff
@@ -303,9 +303,16 @@ private static void ProcessMouseWheelEvent(Sdl2NativeWindow window, MouseWheelEv
...
+
+        private static string PtrToStringAuto(IntPtr ptr)
+        {
+            return SystemDefaultCharSize == 2 ? Marshal.PtrToStringUni(ptr) : Marshal.PtrToStringAnsi(ptr);
```
Comment (VPeruS) [inline src/OpenTK/Platform/SDL2/Sdl2NativeWindow.cs]:
```suggestion
            => IsFilenameUniEncoded ? Marshal.PtrToStringUni(ptr) : Marshal.PtrToStringAnsi(ptr);
```

File: src/OpenTK/Platform/SDL2/Sdl2NativeWindow.cs (hunk 1 above)
Comment (VPeruS) [reply src/OpenTK/Platform/SDL2/Sdl2NativeWindow.cs]: Move it to the beginning of class declaration

**Comment by VPeruS** [comment]
Fixed review comments

**Comment by VPeruS** [comment]
Try to test with utf-8 encoded filenames like "my-𝌆-file"

File: src/osuTK/Platform/SDL2/Sdl2NativeWindow.cs [hunk 3]
```diff
@@ -303,9 +303,12 @@ private static void ProcessMouseWheelEvent(Sdl2NativeWindow window, MouseWheelEv
             window.OnMouseWheel(ev.X, ev.Y, false);
         }
 
+        private static string PtrToStringAuto(IntPtr ptr)
+        => RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? Marshal.PtrToStringUni(ptr) : Marshal.PtrToStringAnsi(ptr);
```
Comment (VPeruS) [inline src/osuTK/Platform/SDL2/Sdl2NativeWindow.cs]:
```suggestion
            => RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? Marshal.PtrToStringUni(ptr) : Marshal.PtrToStringAnsi(ptr);
```

**Comment by smoogipoo** [comment]
Tested on Windows and macOS, seems to work.

**Pull request merged**